controller/backup_target_controller.go

```diff
@@ -546,22 +553,48 @@ func (btc *BackupTargetController) getInfoFromBackupStore(backupTarget *longhorn
 	return info, nil
 }
 
-func (btc *BackupTargetController) syncBackupVolume(backupTarget *longhorn.BackupTarget, backupStoreBackupVolumeNames []string, syncTime metav1.Time, log logrus.FieldLogger) error {
-	backupStoreBackupVolumes := sets.New[string](backupStoreBackupVolumeNames...)
-
+func (btc *BackupTargetController) cleanupDuplicateBackupVolumes(backupTargetName string, log logrus.FieldLogger) (map[string]*longhorn.BackupVolume, error) {
 	// Get a list of all the backup volumes that exist as custom resources in the cluster
-	clusterBackupVolumes, err := btc.ds.ListBackupVolumesWithBackupTargetNameRO(backupTarget.Name)
+	clusterBackupVolumes, err := btc.ds.ListBackupVolumesWithBackupTargetNameRO(backupTargetName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	deletingDuplicateBackupVolumeSet := sets.New[string]()
 	clusterVolumeBVMap := make(map[string]*longhorn.BackupVolume, len(clusterBackupVolumes))
-	clusterBackupVolumesSet := sets.New[string]()
 	for _, bv := range clusterBackupVolumes {
-		clusterBackupVolumesSet.Insert(bv.Spec.VolumeName)
+		if bv.Spec.VolumeName == "" || bv.Spec.BackupTargetName == "" || bv.Spec.BackupTargetName != backupTargetName {
```

Comment:
Is it possible that all the backupVolumes' `spec.VolumeName` or `spec.BackupTargetName` is empty?
What will happen if you delete all the backupVolumes?

Reply:
>Is it possible that all the backupVolumes' spec.VolumeName or spec.BackupTargetName is empty?

I think it might be possible because we still cannot figure out why invalid BackupVolumes had an empty volume name and an empty backup target name

>What will happen if you delete all the backupVolumes?

We only delete the backup volume CRs, and data will still be stored in the backup store.
We can still synchronize the backup information at the next poll interval.